src/coreclr/vm/jithelpers.cpp

```diff
@@ -1788,14 +1788,23 @@ __declspec(selectany) __declspec(thread) uint32_t t_GCThreadStaticBlocksSize;
 __declspec(selectany) __declspec(thread) void** t_NonGCThreadStaticBlocks;
 __declspec(selectany) __declspec(thread) void** t_GCThreadStaticBlocks;
 #else
-EXTERN_C __thread uint32_t t_NonGCMaxThreadStaticBlocks;
-EXTERN_C __thread uint32_t t_GCMaxThreadStaticBlocks;
+struct ThreadStaticBlockInfo
+{
+    uint32_t NonGCMaxThreadStaticBlocks;
+    void** NonGCThreadStaticBlocks;
+
+    uint32_t GCMaxThreadStaticBlocks;
+    void** GCThreadStaticBlocks;
+};
+
+#define t_NonGCMaxThreadStaticBlocks t_ThreadStatics.NonGCMaxThreadStaticBlocks
```

Comment:
I would be nice to change Windows to use `ThreadStaticBlockInfo` too and avoid these defines.